src/components/dialogs/parameters/security-analysis/security-analysis-parameters.tsx

```diff
@@ -5,22 +5,41 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import React, { useCallback, useState } from 'react';
+import React, {
+    ChangeEvent,
+    FunctionComponent,
+    useCallback,
+    useEffect,
+    useState,
+} from 'react';
 import { Grid } from '@mui/material';
 import { DropDown, LabelledButton, styles } from '../parameters.jsx';
 import { LineSeparator } from '../../dialogUtils.jsx';
-import { useIntl } from 'react-intl';
+import { FormattedMessage, useIntl } from 'react-intl';
 import { mergeSx } from '../../../utils/functions.js';
 import {
+    CustomFormProvider,
     DirectoryItemSelector,
     ElementType,
+    SubmitButton,
     useSnackMessage,
 } from '@gridsuite/commons-ui';
 import { fetchSecurityAnalysisParameters } from '../../../../services/security-analysis.js';
-import CreateParameterDialog from '../common/parameters-creation-dialog.tsx';
-import SecurityAnalysisParametersSelector from './security-analysis-parameters-selector.jsx';
+import SecurityAnalysisParametersSelector from './security-analysis-parameters-selector';
+import CreateParameterDialog from '../common/parameters-creation-dialog';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+    getLimitReductionsFormSchema,
+    ILimitReductionsByVoltageLevel,
+    IST_FORM,
+    LIMIT_DURATION_FORM,
+    LIMIT_REDUCTIONS_FORM,
+} from './columns-definitions';
 
-export const SecurityAnalysisParameters = ({ parametersBackend }) => {
+export const SecurityAnalysisParameters: FunctionComponent<
+    Record<string, any>
+> = ({ parametersBackend, setHaveDirtyFields }) => {
```

Comment:
```
FunctionComponent<
    Record<string, any>
>
```
does not match 
`{ parametersBackend, setHaveDirtyFields }`

Reply:
Done